clients/drcachesim/simulator/caching_device_stats.h

```diff
@@ -49,6 +50,20 @@ enum invalidation_type_t {
     INVALIDATION_COHERENCE,
 };
 
+enum metric_name_t {
```

Comment:
I know the existing code has global scope enums but we're trying to avoid that going forward (xref #4343), and here we have very simple names that could conflict.  We do require C++11 so this could be a scoped enum (`typedef enum class metric_name_t`) where it would require the prefix `metric_name_t::HITS`; or at least use a prefix like the existing enums so `METRIC_HITS`; or something, to avoid global symbols with very simple names like `HITS` which could conflict (this code is statically linked into some very large code bases).